src/Form/Form.tsx

```diff
@@ -0,0 +1,198 @@
+                        <button type="button" onClick={handlePrev}>Previous</button>
+                        <button type="button" onClick={handleNext}>Next</button>
+                    </div>
+                );
+            case 4:
+                return (
+                    <div>
+                        <h3>Savings</h3>
+                        {budgetInfo.expenses.savings.map((expense, index) => (
+                            <div key={index}>
+                                <label>Name</label>
+                                <input
+                                    type="text"
+                                    name="name"
+                                    value={expense.name}
+                                    onChange={(e) => handleChange(e, 'savings', index)}
+                                />
+                                <label>Amount</label>
+                                <input
+                                    type="number"
+                                    name="amount"
+                                    value={expense.amount}
+                                    onChange={(e) => handleChange(e, 'savings', index)}
+                                />
+                            </div>
+                        ))}
+                        <button type="button" onClick={() => handleAddExpense('savings')}>Add Saving</button>
+                        <button type="button" onClick={handlePrev}>Previous</button>
+                        <button type="submit">Submit</button>
+                    </div>
+                );
+            default:
+                return null;
+        }
+    };
+
+    const transformDataForChart = () => {
+        const { wants, needs, savings } = budgetInfo.expenses;
+        const totalWants = wants.reduce((sum, item) => sum + item.amount, 0);
+        const totalNeeds = needs.reduce((sum, item) => sum + item.amount, 0);
+        const totalSavings = savings.reduce((sum, item) => sum + item.amount, 0);
+        
+        return {
+            labels: ['Needs', 'Wants', 'Savings'],
+            datasets: [
+                {
+                    label: 'Expenses',
+                    data: [totalNeeds, totalWants, totalSavings],
+                    backgroundColor: [
+                        "rgba(19, 141, 117, 0.9)",
+                        "rgba(84, 153, 199, 0.9)",
+                        "rgba(91, 124, 153, 0.9)"
+                    ],
+                    hoverOffset: 4
+                }
+            ]
+        };
+    };
+
+    return (
+        <>
+        <div className="container">
+            <div className="budget">
+                    <form className="form" onSubmit={handleSubmit}>
+                        {renderStep()}
+                    </form>
+            </div>
+            <div className="pie-chart">
+                <PieChart data={transformDataForChart()} />
+            </div>
+        </div>
+        </>
+    );
+};
+
+export default Form
```

Comment:
This is so much repetitive work. I hope you copy pasta'd pieces of this to save time. Holy 198 lines.